src/components/molecules/MapPartygoerOverlay/MapPartygoerOverlay.tsx

```diff
@@ -26,47 +31,28 @@ export const MapPartygoerOverlay: React.FC<MapPartygoerOverlayProps> = ({
   withMiniAvatars = false,
 }) => {
   const isMe = partygoer.id === myUserUid;
-  const position = partygoer?.data?.[venueId];
-  const currentRow = position?.row ?? 0;
-  const currentCol = position?.column ?? 0;
-  const avatarWidth = 100 / totalColumns;
-  const avatarHeight = 100 / totalRows;
 
-  const containerStyle = useMemo(
-    () => ({
-      display: "flex",
-      width: `${avatarWidth}%`,
-      height: `${avatarHeight}%`,
-      position: "absolute",
-      cursor: "pointer",
-      transition: "all 1400ms cubic-bezier(0.23, 1 ,0.32, 1)",
-      top: `${avatarHeight * (currentRow - 1)}%`,
-      left: `${avatarWidth * (currentCol - 1)}%`,
-      justifyContent: "center",
-      zIndex: "11",
-    }),
-    [avatarHeight, avatarWidth, currentCol, currentRow]
-  );
+  const { row = 0, column = 0 } = partygoer?.data?.[venueId] ?? {};
+
+  const containerVars = useCss({
+    "--map-partygoer-overlay-total-rows": totalRows,
+    "--map-partygoer-overlay-total-columns": totalColumns,
+    "--map-partygoer-overlay-row": row,
+    "--map-partygoer-overlay-column": column,
+  });
 
-  const pictureUrl = partygoer?.pictureUrl;
-  const avatarStyle = useMemo(
-    () => ({
-      width: "80%",
-      height: "80%",
-      borderRadius: "100%",
-      alignSelf: "center",
-      backgroundImage: `url(${pictureUrl})`,
-      backgroundSize: "cover",
-    }),
-    [pictureUrl]
+  const containerClasses = classNames(
+    "MapPartygoerOverlay__avatar-container",
+    containerVars
   );
```

Comment:
[FYI] Calling out this pattern of using CSS `--variables` to pass dynamic content from the JS into SCSS, so that we can keep our styles in SCSS rather than inline styles.

(cc // @sparkletown/engineering FYI: you'll likely see more of this pattern moving forward)